### PR TITLE
Virtual python environment in setup

### PR DESCRIPTION
**Pull request by kuthedk**
## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] My code follows the style guidelines enforced by static analysis tools.
- [x] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.
- [x] Will this be part of a product update? If yes, please write one phrase about this update.

## Pull Request Details
Wrike Ticket - [Investigate setup.sh script creating a virtual python environment that could be used and update the documentation on using it](https://www.wrike.com/open.htm?id=1033377805)

## Changes made

setup.sh:
- A command is added to navigate to the backend directory
- A command is added to create a python virtual environment named `rosalution_env`
- A command is added to activate the virtual environment
- A command is added to install requirements.txt in the virtual environment
- A command is added to deactivate the virtual environment
- A command is added to navigate back to the root directory and continue the script.

README.md:
- updated the documentation to reflect the changes to the `setup.sh` script

gitignore:
- added the folder created for the venv to the gitignore

## **To Review:**

- [ ] Static Analysis by Reviewer
- [ ] Check the documentation is accurate.
- [ ] The changes made to the `setup.sh` script are working as intended.
  To check this run the following commands:

```bash
./setup.sh
```

- [ ] All Github Actions checks have passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,10 +75,12 @@ The script will
 - `yarn install` within each subdirectory
 - Updates your local `/etc/hosts` to support the local DNS name redirect
 'local.rosalution.cgds' to localhost.
+- creates a virtual environment for Python called "rosalution_env" within the backend directory and activates it.
+- installs Python dependencies within the virtual environment and then deactivates the virtual environment.
+    - use `source backend/rosalution_env/bin/activate` to activate the virtual environment.
```
Comment (SeriousHorncat) [inline README.md:80]: You can move this line to be in the Python README.  

We can also add an echo output in the script notifying the user that they can activate the virtual environment via `source backend/rosalution_env/bin/activate`.

File: backend/README.md [hunk 2]
```diff
@@ -15,7 +15,9 @@ and the web accessible Swagger API documentation.
 ### Requirements
 
 Project dependencies are listed in the `requirements.txt` file at the project's
-root structure.
+backend root structure.
+
+you do not need to run this command if you have already ran the ./setup.sh script in the Rosalution root directory.
```
Comment (SeriousHorncat) [inline backend/README.md:20]: This and the example below can be revised to discuss how to activate the python venv to run the python tools.

Comment (kuthedk) [reply backend/README.md:20]: I think I have made the requested change.

File: README.md (hunk 1 above)
Comment (kuthedk) [reply README.md]: I believe I have now made the requested changes.

File: backend/README.md (hunk 2 above)
Comment (JmScherer) [reply backend/README.md:20]: I think if someone wanted to install the python virtual environment without using the setup.sh script, those commands should be outlined in the readme where the original `pip install -r requirements.txt` is.

Comment (JmScherer) [reply backend/README.md:20]: After trying out the virtual environment for python, perhaps a more detailed explanation how the virtual environment works as Ange is pointing out. 

For example, the setup.sh script creates a virtual environment for a single terminal session. Once it closes, then the terminal no longer has knowledge of the virtual environment and needs to be re-sourced each time you work with the python tools in the backend.

Comment (SeriousHorncat) [reply backend/README.md:20]: We can probably circle back on how we want to document and tidy things up.  This works for getting things started for now and we can move on to some other repo cleanup work.

**Review by SeriousHorncat — APPROVED** [review]

**Pull request merged**